app/models/answer.rb

```diff
@@ -43,22 +44,16 @@ def self.location_answers_for_mission(mission, user = nil, options = {})
     end
 
     # return an AR relation
-    joins(:response, questioning: :question)
-      .joins(%{LEFT JOIN `option_sets` ON `questions`.`option_set_id` = `option_sets`.`id`})
-      .joins(%{LEFT JOIN `options` ON `option_sets`.`allow_coordinates` = 1 AND `questions`.`qtype_name` = 'select_one' AND `answers`.`option_id` = `options`.`id`})
-      .joins(%{LEFT JOIN `choices` ON `option_sets`.`allow_coordinates` = 1 AND `questions`.`qtype_name` = 'select_multiple' AND `answers`.`id` = `choices`.`answer_id`})
-      .joins(%{LEFT JOIN `options` AS `choice_options` ON `choices`.`option_id` = `choice_options`.`id`})
+    joins(:response)
+      .joins(%{LEFT JOIN `choices` ON `choices`.`answer_id` = `answers`.`id`})
       .where(:responses => response_conditions)
       .where(%{
-        (`questions`.`qtype_name` = 'location' AND `answers`.`value` IS NOT NULL)
-        OR (`options`.`latitude` IS NOT NULL AND `options`.`longitude` IS NOT NULL)
-        OR (`choice_options`.`latitude` IS NOT NULL AND `choice_options`.`longitude` IS NOT NULL)
+        (`answers`.`latitude` IS NOT NULL AND `answers`.`longitude` IS NOT NULL)
+        OR (`choices`.`latitude` IS NOT NULL AND `choices`.`longitude` IS NOT NULL)
       })
       .select(:response_id,
-        %{COALESCE(`answers`.`value`,
-                   CONCAT(`options`.`latitude`, ' ', `options`.`longitude`),
-                   CONCAT(`choice_options`.`latitude`, ' ', `choice_options`.`longitude`)) AS `value`})
-      .distinct
+        %{COALESCE(CONCAT(`answers`.`latitude`, ' ', `answers`.`longitude`),
+                   CONCAT(`choices`.`latitude`, ' ', `choices`.`longitude`)) AS `value`})
```

Comment:
Yep, that part wasn't changed to avoid possible collisions with old code. I'll add the commentary explaining that this should be temporary.

About the dashboard, I'm not sure i follow. This query is the one being used on the dashboard. Are we talking about the same thing?